crates/op-rbuilder/src/tx.rs

```diff
@@ -0,0 +1,250 @@
+use std::sync::Arc;
+
+use alloy_consensus::{
+    conditional::BlockConditionalAttributes, BlobTransactionSidecar, BlobTransactionValidationError,
+};
+use alloy_eips::{eip7702::SignedAuthorization, Typed2718};
+use alloy_primitives::{Address, Bytes, TxHash, TxKind, B256, U256};
+use alloy_rpc_types_eth::{erc4337::TransactionConditional, AccessList};
+use reth_optimism_primitives::OpTransactionSigned;
+use reth_optimism_txpool::{
+    conditional::MaybeConditionalTransaction, estimated_da_size::DataAvailabilitySized,
+    interop::MaybeInteropTransaction, OpPooledTransaction,
+};
+use reth_primitives::{kzg::KzgSettings, Recovered};
+use reth_primitives_traits::InMemorySize;
+use reth_transaction_pool::{EthBlobTransactionSidecar, EthPoolTransaction, PoolTransaction};
+
+pub trait FBPoolTransaction:
```

Comment:
Suggestion:
Create trait MaybeRevertingTransaction (without bounds)
Create supertrait FBPoolTransaction:     EthPoolTransaction + MaybeInteropTransaction + MaybeConditionalTransaction + DataAvailabilitySized + MaybeRevertingTransaction
Implement MaybeRevertingTransaction for FBPooledTransaction